cmd/collect_signals/github/collector.go

```diff
@@ -0,0 +1,120 @@
+	ghr, ok := r.(*repo)
+	if !ok {
+		return nil, errors.New("project is not a github project")
+	}
+	now := time.Now()
+
+	s := &signal.RepoSet{
+		URL:          signal.Val(r.URL().String()),
+		Language:     signal.Val(ghr.BasicData.PrimaryLanguage.Name),
+		License:      signal.Val(ghr.BasicData.LicenseInfo.Name),
+		StarCount:    signal.Val(ghr.BasicData.StargazerCount),
+		CreatedAt:    signal.Val(ghr.createdAt()),
+		CreatedSince: signal.Val(legacy.TimeDelta(now, ghr.createdAt(), legacy.SinceDuration)),
+		UpdatedAt:    signal.Val(ghr.updatedAt()),
+		UpdatedSince: signal.Val(legacy.TimeDelta(now, ghr.updatedAt(), legacy.SinceDuration)),
+		// Note: the /stats/commit-activity REST endpoint used in the legacy Python codebase is stale.
+		CommitFrequency: signal.Val(legacy.Round(float64(ghr.BasicData.DefaultBranchRef.Target.Commit.RecentCommits.TotalCount)/52, 2)),
+	}
+	ghr.logger.Debug("Fetching contributors")
+	if contributors, err := legacy.FetchTotalContributors(ctx, ghr.client, ghr.owner(), ghr.name()); err != nil {
+		return nil, err
+	} else {
+		s.ContributorCount.Set(contributors)
+	}
+	ghr.logger.Debug("Fetching org count")
+	if orgCount, err := legacy.FetchOrgCount(ctx, ghr.client, ghr.owner(), ghr.name()); err != nil {
+		return nil, err
+	} else {
+		s.OrgCount.Set(orgCount)
+	}
+	ghr.logger.Debug("Fetching releases")
+	if releaseCount, err := legacy.FetchReleaseCount(ctx, ghr.client, ghr.owner(), ghr.name(), legacyReleaseLookback); err != nil {
+		return nil, err
+	} else {
+		if releaseCount != 0 {
+			s.RecentReleaseCount.Set(releaseCount)
+		} else {
+			timeSinceCreated := int(now.Sub(ghr.createdAt()).Hours()) / 24
+			t := (ghr.BasicData.Tags.TotalCount * legacyReleaseLookbackDays) / timeSinceCreated
+			s.RecentReleaseCount.Set(t)
+		}
+	}
+	return s, nil
+}
+
+func (rc *RepoCollector) IsSupported(p projectrepo.Repo) bool {
+	_, ok := p.(*repo)
+	return ok
+}
+
+type IssuesCollector struct {
+}
+
+func (ic *IssuesCollector) EmptySet() signal.Set {
+	return &signal.IssuesSet{}
+}
+
+func (ic *IssuesCollector) Collect(ctx context.Context, r projectrepo.Repo) (signal.Set, error) {
+	ghr, ok := r.(*repo)
+	if !ok {
+		return nil, errors.New("project is not a github project")
+	}
+	s := &signal.IssuesSet{}
+
+	// TODO: these should cache on the project
+	ghr.logger.Debug("Fetching updated issues")
+	if up, err := legacy.FetchIssueCount(ctx, ghr.client, ghr.owner(), ghr.name(), legacy.IssueStateAll, legacy.IssueLookback); err != nil {
+		return nil, err
+	} else {
+		s.UpdatedCount.Set(up)
+		if up != 0 {
+			ghr.logger.Debug("Fetching comment frequency")
+			if comments, err := legacy.FetchIssueCommentCount(ctx, ghr.client, ghr.owner(), ghr.name(), legacy.IssueLookback); err != nil {
+				if errors.Is(err, legacy.TooManyResultsError) {
+					ghr.logger.Debug("Comment count failed with too many result")
+					s.CommentFrequency.Set(2.0)
```

Comment:
nit: put 2.0 in a constant to make it clear this is the default? 

Also, why 2.0 as the default?

Reply:
2.0 was roughly what I was seeing on repos (usually between 1.5 and 2.5) for comment frequency.

It is a bit of a magic value, and leaving it unset may be more accurate.

Reply:
ack.